File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        hashmap = {}
        for n in nums:
            hashmap[n] = hashmap.get(n, 0) + 1

        items = sorted(hashmap.items(), key= lambda x: x[1], reverse=True)
        return [num for num, item in items[:k]]

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] prod = new int[nums.length];
        prod[0] = 1;

        for(int i = 1; i < nums.length; i++){
            prod[i] = prod[i-1] * nums[i-1];
        }

        int suffix = 1;
        for(int i = nums.length - 2; i >= 0; i--){
            suffix *= nums[i+1];
            prod[i] *= suffix;
        }

        return prod;
    }
}  
